Tests/MapboxMapsTests/Foundation/Camera/FlyToTests.swift

```diff
@@ -59,4 +60,66 @@ internal class FlyToTests: XCTestCase {
         XCTAssertEqual(d2.longitude, s2.longitude, accuracy: 0.00001)
     }
     // swiftlint:enable identifier_name
+
+    func testForValidValues() {
+
+        let epsilon: CLLocationDegrees = 0.00001
+
+        for _ in 0..<10 {
+
+            // Longitude increases easterly for this test.
+            let sourceCoord = CLLocationCoordinate2D(
+                latitude: CLLocationDegrees.random(in: -85..<85),
+                longitude: CLLocationDegrees.random(in: 0..<180)
+            )
+
+            let destCoord = CLLocationCoordinate2D(
+                latitude: CLLocationDegrees.random(in: -85..<85),
+                longitude: CLLocationDegrees.random(in: 180..<360)
+            )
+
+            let source = CameraOptions(center:sourceCoord,
+                                       padding: .zero,
+                                       zoom: 14,
+                                       bearing: 0,
+                                       pitch: 0)
+            let dest = CameraOptions(center: destCoord,
+                                     zoom: 18,
+                                     bearing: 90,
+                                     pitch: 45)
+
+            guard let flyTo = FlyToInterpolator(from: source, to: dest, size: CGSize(width: 500.0, height: 500.0)),
+                  var boundingBox = BoundingBox(from: [sourceCoord, destCoord]) else {
+                XCTFail("Failed to create interpolator")
+                continue
+            }
+
+            boundingBox.southWest.latitude -= epsilon
+            boundingBox.southWest.longitude -= epsilon
+            boundingBox.northEast.latitude += epsilon
+            boundingBox.northEast.longitude += epsilon
+
+            let duration = flyTo.duration()
+            XCTAssert(duration > 0)
+
+            for t: Double in stride(from: 0, to: 1, by: 0.01) {
+
+                let coordinate = flyTo.coordinate(at: t)
+
+                XCTAssert(boundingBox.contains(coordinate, ignoreBoundary: false), "t=\(t) coordinate=\(coordinate) boundingBox=\(boundingBox)")
+
+                // Zoom doesn't go below start or end
+                let zoom = CGFloat(flyTo.zoom(at: t))
+                XCTAssert(zoom <= max(source.zoom!, dest.zoom!), "t=\(t) zoom=\(zoom)")
+
+                let bearing = CLLocationDirection(flyTo.bearing(at: t))
+                XCTAssert(bearing >= source.bearing!, "t=\(t) bearing=\(bearing)")
```

Comment:
We could use a range here to make this cleaner.

Reply:
Just tried, and it ends up looking worse. Let's tackle later if it becomes an issue.